File: content/showcase/7-type-reflection/type-reflection.zig
// @TypeOf reads a type at compile time; types compare
// with ==
const std = @import("std");

pub fn main() void {
    const x: i32 = 5;
    const same = @TypeOf(x) == i32; // true
    const cross = @TypeOf(x) == f64; // false
    const meta = @TypeOf(i32) == type; // true
    const all = same and meta and !cross;
    std.debug.print("{}\n", .{all});
}

File: content/showcase/7-type-reflection/type-reflection.rs
// a value's type has an identity at run time,
// though nothing more of it can be read back
use std::any::{Any, TypeId};

fn main() {
    let x: i32 = 5;
    let same = x.type_id() == TypeId::of::<i32>();
    let cross = x.type_id() == TypeId::of::<f64>();
    println!("{}", same && !cross); // true
}
